partner_email_check/models/res_partner.py

```diff
@@ -24,14 +24,24 @@
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
+    def copy_data(self, default=None):
+        res = super(ResPartner, self).copy_data(default=default)
+        if self._should_filter_duplicates():
+            for copy_vals in res:
+                if "email" in copy_vals:
+                    copy_vals["email"] = ""
```

Comment:
Why not delete the value from the dictionary, or set it to False, so email will have a NULL value in the database.